File: DSA/done/3396 - minimumOperations.py
# https://leetcode.com/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/

from typing import List

class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        seen = [False] * 101
        for i in range(len(nums) - 1, -1, -1):
            if seen[nums[i]]:
                return i // 3 + 1
            seen[nums[i]] = True
        return 0

File: DSA/done/3396 - minimumOperations.java
// https://leetcode.com/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/

class Solution {
    public int minimumOperations(int[] nums) {
        boolean[] seen = new boolean[101];
        for (int i = nums.length - 1; i > -1; i--) {
            if (seen[nums[i]]) {
                return i / 3 + 1;
            }
            seen[nums[i]] = true;
        }
        return 0;
    }
}